superset-frontend/src/views/CRUD/data/dataset/AddDataset/DatasetPanel/DatasetPanel.tsx

```diff
@@ -168,15 +202,52 @@ export interface IDatasetPanelProps {
    * Boolean indicating if the component is in a loading state
    */
   loading: boolean;
+  datasets?: DatasetObject[] | undefined;
 }
 
+const EXISTING_DATASET_DESCRIPTION = t(
+  'You can only associate one dataset with one table. This table already has a dataset associated with it in Preset.\n',
```

Comment:
Hmm, does this need to be preset specific language? If so, then I think we need to add that in on the Preset side and have a more generic description in superset.

Reply:
I'm not sure about Preset-specific language here, I think we'd need @yousoph 's input on this one.

Reply:
Updated text: This table already has a dataset associated with it. You can only associate one dataset with a table.

Reply:
I spoke with Sophie and Karan in Slack and we discussed changing the text to be more general and not include Superset or Preset. Changed in [`this commit`](https://github.com/apache/superset/pull/22136/commits/f4a42c14eca0c2b1283a0b9b68cc27e31cc2e2d3).